### PR TITLE
Initialize AndroidSurfaceGL and its associated resource context before ANativeWindow acquisition.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This allows us to create the IO thread context way earlier and service load requests on the same. Before this patch, early lifecycle loads would use the default Skia texture loader instead of the Async texture loader we provide.

Fixes https://github.com/flutter/flutter/issues/6581

### CODE REVIEW

File: shell/platform/android/android_surface_gl.cc [hunk 1]
```diff
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include <utility>
 #include "flutter/shell/platform/android/android_surface_gl.h"
```
Comment (abarth) [inline shell/platform/android/android_surface_gl.cc]: Usually we have a blank line after the primary header.

File: shell/platform/android/android_surface_gl.cc [hunk 2]
```diff
@@ -2,8 +2,8 @@
...
+#include <utility>
```
Comment (abarth) [inline shell/platform/android/android_surface_gl.cc]: Usually we have a blank line between libc++ headers and project headers.

**Comment by chinmaygarde** [comment]
Now that `AndroidSurfaceGL` is long lived, we can also get rid of the static environment that holds onto the EGL display connection. Will clean that up in a separate patch.

**Comment by abarth** [comment]
LGTM

Do we not have similar problems on other platforms?

File: shell/platform/android/android_surface_gl.cc (hunk 1 above)
Comment (chinmaygarde) [inline shell/platform/android/android_surface_gl.cc]: Clang format applies the header ordering rules automatically now. But it decides to stop if we introduce a space anywhere. This way, we can just plop includes anywhere we want and let Clang format do its thing correctly.

I didn't notice anything in https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes about spaces but I am happy to add the same if this is an unwritten rule.

**Comment by chinmaygarde** [comment]
> Do we not have similar problems on other platforms?

iOS sets up the `IOSGLContext` in the ` PlatformViewIOS` constructor and immediately calls `SetupResourceContextOnIOThread`. So this was never on issue there. Other platforms never have had resource loading contexts. So there is nothing to do there.

File: shell/platform/android/android_surface_gl.cc [hunk 3]
```diff
@@ -40,15 +40,28 @@ static ftl::RefPtr<AndroidContextGL> GlobalResourceLoadingContext(
...
-                                   PlatformView::SurfaceConfig offscreen_config)
-    : valid_(false) {
+AndroidSurfaceGL::AndroidSurfaceGL(
+    PlatformView::SurfaceConfig offscreen_config) {
   // Acquire the offscreen context.
   offscreen_context_ = GlobalResourceLoadingContext(offscreen_config);
 
   if (!offscreen_context_ || !offscreen_context_->IsValid()) {
-    return;
+    offscreen_context_ = nullptr;
```
Comment (chinmaygarde) [inline shell/platform/android/android_surface_gl.cc]: There is also an `IsValid()` check there.

**Pull request merged**